### PR TITLE
Restore existing files mocked during the gulp test process

### LINKED ISSUE #596: config.local.json is modified by some tests
## Issue Overview

When we run `npm run test:gulp:prod:setup`, `config.local.json` is create or modified. As `config.local.json` has precedence over other config files, `npm run bundle` will produce prod files from unexpected settings.

## Steps to Reproduce

1. Create a `config.local.json` with custom values in `config/`
2. Run `npm run test:gulp:prod:setup`
3. See that `config.local.json` has been modified

Content of `config.local.json` after `npm run test:gulp:prod:setup`
```json
{
  "theme": {
    "slug": "gulp-test-prod-theme",
    "name": "Gulp Test Prod Theme",
    "author": "Gulp Bot",
    "PHPNamespace": "Gulp_Bot\\Theme"
  }
}

```

## Expected Behavior

Tests should not modify content or config files.

## Current Behavior

The content of `config.local.json` is modified by `npm run test:gulp:prod:setup`.

### PR DESCRIPTION
**Pull request by ataylorme**
## Description

Closes #596. If a file already exists in the mock file destination rename it with a suffix of `-existing` during gulp prod test set up and rename it back to the original filename during gulp prod test cleanup.

## List of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] This pull request relates to a ticket.
- [x] This code is tested.
- [ ] This change has been added to CHANGELOG.md
- [x] I want my code added to WP Rig.

### CODE REVIEW

File: gulp/tests/prod-build/prod-build.setup.js [hunk 1]
```diff
@@ -13,5 +13,14 @@ import { filesToMock } from './prod-build.utils';
 
 // Copy the mock files to their destination before testing.
 filesToMock.forEach( ( file ) => {
+	// Check if the file already exists
+	const fileExists = fs.existsSync( file.dest );
+	if ( fileExists ) {
+		// If it does, rename the file adding an -existing suffix
+		const existingFile = file.dest.replace( /(\.[\w\d_-]+)$/i, '-existing$1' );
```
Comment (benoitchantre) [inline gulp/tests/prod-build/prod-build.setup.js]: This regexp could be simplified like this: `/(\.[\w-]+)$/i`
`\w` matches `[a-zA-Z0-9_]`, we don't need `\d` and `_`.

File: gulp/tests/prod-build/prod-build.teardown.js [hunk 2]
```diff
@@ -18,9 +18,18 @@ import {
 
 // Delete the mock files after testing.
 filesToMock.forEach( ( file ) => {
+	// Delete the mock file
 	if ( fs.existsSync( file.dest ) ) {
 		fs.unlinkSync( file.dest );
 	}
+
+	// Check if a previous version of the file exists
+	const existingFile = file.dest.replace( /(\.[\w\d_-]+)$/i, '-existing$1' );
```
Comment (benoitchantre) [inline gulp/tests/prod-build/prod-build.teardown.js]: This regexp could be simplified like this: `/(\.[\w-]+)$/i`
`\w` matches `[a-zA-Z0-9_]`, we don't need `\d` and `_`.

Comment (ataylorme) [reply gulp/tests/prod-build/prod-build.teardown.js]: @benoitchantre updated

File: gulp/tests/prod-build/prod-build.setup.js (hunk 1 above)
Comment (ataylorme) [reply gulp/tests/prod-build/prod-build.setup.js]: @benoitchantre updated

File: gulp/tests/prod-build/prod-build.teardown.js (hunk 2 above)
Comment (benoitchantre) [reply gulp/tests/prod-build/prod-build.teardown.js]: Thank you for the changes.

**Comment by benoitchantre** [comment]
@ataylorme Can you resolve the conflict?

**Pull request merged**